### DISCUSSION TITLE
How can I import a Nanobind module into a Python script

### DISCUSSION THREAD

**Discussion by samh133** [wjakob/nanobind#762] [Q&A]
I have created a module and it seems to import and run with no problems when I start a bash shell in the build folder,  however I do not seem to be able to import it with a Python script. 

This is my current import statement which I found from this https://www.geeksforgeeks.org/python-import-module-from-different-directory/

```
import return_thermal_data

import sys
sys.path.insert(0, '/home/user/get_thermal_values/build')
```

However I get the error "Unable to import 'return_thermal_data'" I have also attempted to place the Python file directly in the build folder but this has not worked either.

Here is the nanobind_add_module statement from CMakeLists.txt

`nanobind_add_module(return_thermal_data return_thermal_data.cpp)`

And here is the NB_MODULE statement in the CSS

```
NB_MODULE(return_thermal_data, m) {
m.def("subscribe_to_topic", &subscribeToTopic, "Subscribe to the data topic");}

```
As far as I can tell the Nanobind docs only demo using modules via the Python shell. Is it possible to use them with Pythons scripts, or is there another tools I should use to achieve this?

**Comment by wojdyr** [comment]
I tried how ChatGPT and Claude would respond to this. Claude did better here.

![image](https://github.com/user-attachments/assets/b8024dc6-9356-4865-94a6-5ba8728f8966)

The full answer had 5 points, but I think it nailed it in the first one.
